internal/handler/task.go

```diff
@@ -111,287 +175,121 @@ func (h *Handler) ListCmd() *cobra.Command {
-		return &d, nil
-	default:
-		for _, v := range dateFormats {
-			t, err := time.Parse(v, arg)
-			if err == nil {
-				return &t, nil
+			if len(ids) > 1 {
+				fmt.Printf("Finished %d tasks.\n", len(ids))
+			} else {
+				fmt.Printf("Finished 1 task.\n")
 			}
-		}
-		return nil, fmt.Errorf("invalid date format: %s", arg)
+		},
 	}
+	return doneCmd
 }
 
-func parseTimeStamp(arg string) (*time.Time, *time.Time, error) {
-	hour, colon, minute, am, dash := false, false, false, false, false
-	startHour, endHour, startMinute, endMinute := -1, -1, -1, -1
-	startFormat, endFormat := "", ""
-
-	for i := 0; i < len(arg); i++ {
-		switch arg[i] {
-		case ':':
-			if !hour || minute {
-				return nil, nil, fmt.Errorf("colon is expected after hour, not minute: %s", arg)
-			} else if am {
-				return nil, nil, fmt.Errorf("colon can never occur after 'am' or 'pm': %s", arg)
-			} else if colon {
-				return nil, nil, fmt.Errorf("colons cannot be duplicated for same hour, minute combination: %s", arg)
-			}
-			colon = true
-		case '-':
-			if !hour || dash {
-				return nil, nil, fmt.Errorf("dashes require an hour and cannot be duplicated: %s", arg)
-			}
-			dash = true
-			hour = false
-			colon = false
-			minute = false
-			am = false
-		case 'a', 'p':
-			if !hour || am {
-				return nil, nil, fmt.Errorf("'am'/'pm' cannot occur without an hour nor can there be duplicates: %s", arg)
+func (h *Handler) UndoCmd() *cobra.Command {
+	undoCmd := &cobra.Command{
+		Use:   "undo",
+		Short: "Marks task as incomplete by ID",
+		Run: func(cmd *cobra.Command, args []string) {
+			ids, err := parseDone(args)
+			if err != nil {
+				log.Fatal(err)
 			}
-			if i+1 < len(arg) && arg[i+1] == 'm' {
-				s := arg[i : i+2]
-				if !dash {
-					startFormat = s
-				} else {
-					endFormat = s
+			for _, i := range ids {
+				t, err := h.service.GetTask(i)
+				if err != nil {
+					log.Fatal(err)
 				}
-			} else {
-				return nil, nil, fmt.Errorf("provided time tag besides valid 'am'/'pm': %s", arg)
-			}
-			am = true
-			i++
-		case '0', '1', '2', '3', '4', '5', '6', '7', '8', '9':
-			if hour && !colon {
-				return nil, nil, fmt.Errorf("minutes must be separated by colon from hours: %s", arg)
-			} else if minute {
-				return nil, nil, fmt.Errorf("minutes cannot be duplicated: %s", arg)
-			} else if am {
-				return nil, nil, fmt.Errorf("hours and minutes cannot come after 'am'/'pm' signature: %s", arg)
-			} else if hour {
-				minute = true
-				if i+1 < len(arg) && arg[i+1] >= '0' && arg[i+1] <= '9' {
-					x, _ := strconv.Atoi(arg[i : i+2])
-					if x > 59 {
-						return nil, nil, fmt.Errorf("minutes cannot be greater than 59: %s", arg)
-					}
-					if !dash {
-						startMinute = x
-					} else {
-						endMinute = x
-					}
-					i++
-				} else {
-					return nil, nil, fmt.Errorf("minutes require 2 digits: %s", arg)
+				//check if task exists, because of nil pointer dereference
+				if t == nil {
```

Comment:
We have to check, if task is not nil, because it will always throw "panic: runtime error: invalid memory address or nil pointer dereference"